orderfile/app/src/main/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.18.1)
+project(OrderfileDemo CXX)
+
+set(GENERATE_PROFILES ON)
+#set(USE_PROFILE "${CMAKE_SOURCE_DIR}/demo.orderfile")
+
+add_library(orderfiledemo SHARED orderfile.cpp)
+target_link_libraries(orderfiledemo log)
+
+if(GENERATE_PROFILES)
+    target_compile_options(orderfiledemo PRIVATE -forder-file-instrumentation -O1 -mllvm -orderfile-write-mapping=mapping.txt )
```

Comment:
If `-O1` specifically is required rather than just not `-O0`, this deserves a comment (worth explaining in the readme as well).

If I could think of a good way to write it, it'd maybe be better to check the current optimization level and just fail the build for incompatible settings? CMake optimization flags are applied in strange and mysterious ways and it's usually best to just let cmake drive. What you have is probably the best bet for a sample though?

Reply:
Still todo, otherwise LGTM